mnefun/_mnefun.py

```diff
@@ -842,6 +897,93 @@ def run_sss_positions(fname_in, fname_out, host='kasga', opts='', port=22):
     run_subprocess(cmd)
 
 
+def run_sss_localy(p, subjects, run_indices):
+    """Run SSS locally using maxwell filter in python
+
+    .. warning:: Automatic bad channel detection is not currently implemented.
+                 It is critical to mark bad channels before running Maxwell
+                 filtering, so data should be inspected and marked accordingly
+                 prior to running this algorithm.
+
+    .. warning:: Not all features of Elekta MaxFilter™ are currently
+                 implemented (see Notes). Maxwell filtering in mne-python
+                 is not designed for clinical use.
+
+    Notes
+    -----
+    Compared to Elekta's MaxFilter™ software, Maxwwell filtering in mne-python
+    does not require/support the following arguments:
+        -format
+        -hpicons
+        -autobad
+        -force
+
+    """
+    cal_file = op.join(op.dirname(op.dirname(__file__)), 'mnefun/data/sss_cal.dat')
+    ct_file = op.join(op.dirname(op.dirname(__file__)), 'mnefun/data/ct_sparse.fif')
+    if p.tsss_dur:
+        st_duration = p.tsss_dur
+    else:
+        st_duration = None
+    for si, subj in enumerate(subjects):
+        if p.disp_files:
+            print('  Maxwell filtering subject %g/%g (%s).'
+                  % (si + 1, len(subjects), subj))
+        #  locate raw and erm files
+        sss_dir = op.join(p.work_dir, subj, p.sss_dir)
+        if not op.isdir(sss_dir):
+            os.mkdir(sss_dir)
+        raw_files = get_raw_fnames(p, subj, 'raw', False, False, run_indices[si])
+        raw_files_out = get_raw_fnames(p, subj, 'sss', False, False, run_indices[si])
+        erm_files = get_raw_fnames(p, subj, 'raw', 'only')
+        erm_files_out = get_raw_fnames(p, subj, 'sss', False, False, run_indices[si])
+        prebad_file = _prebad(p, subj)
+        #  process raw files
+        for ii, (r, o) in enumerate(zip(raw_files, raw_files_out)):
+            if not op.isfile(r):
+                raise NameError('File not found (' + r + ')')
+            raw = Raw(r, preload=True, allow_maxshield=True)
+            raw.load_bad_channels(prebad_file, force=True)
```

Comment:
Need to do `raw.fix_mag_coil_types()` call. Probably also need `mne.chpi.filter_chpi(raw)`, too.